### PR TITLE
_get_nested_attr should return a value in the general case

### LINKED ISSUE #1053: _get_nested_attr returning None due to missing a return in the general case
I noticed that the `old_state` variable of  `make_functional.py::FunctionalModuleWithBuffers::forward` is always full of `None` elements. Digging a bit I saw that there is no return in the general case of [`_get_nested_attr`](https://github.com/pytorch/pytorch/blob/91b3cd0b5a7d297a82ca0f9068ea7f9ac1963ced/functorch/_src/make_functional.py#L43)

```def _get_nested_attr(obj: nn.Module, names: List[str]) -> None:
    if len(names) == 1:
        return getattr(obj, names[0])
    else:
        # No return here
        _get_nested_attr(getattr(obj, names[0]), names[1:])
```

So the outermost call is always returning `None` when `len(names) > 1`. If I'm right and this is a bug what implications does it have? For instance, the "Remove the loaded state on self.stateless_model" step surely wouldn't be doing what it's supposed to.

### PR DESCRIPTION
**Pull request by srossross**
Fixes https://github.com/pytorch/functorch/issues/1053

### CODE REVIEW

**Comment by srossross** [comment]
tested on this script

```
import torch
import torch.nn as nn
from functorch import make_functional
x = torch.randn(4, 3)


class NeuralNetwork(nn.Module):
    def __init__(self):
        super(NeuralNetwork, self).__init__()
        self.flatten = nn.Flatten()
        self.linear_relu_stack = nn.Sequential(
            nn.Linear(3,3),
            nn.ReLU(),
            nn.Linear(3, 3),
            nn.ReLU(),
            nn.Linear(3, 3),
        )

    def forward(self, x):
        x = self.flatten(x)
        logits = self.linear_relu_stack(x)
        return logits

model = NeuralNetwork()

func, params = make_functional(model)
func(params, x)

print("func.stateless_model.linear_relu_stack[0].weight", func.stateless_model.linear_relu_stack[0].weight)
```

## before:
```
func.stateless_model.linear_relu_stack[0].weight None
```

## after:
```
func.stateless_model.linear_relu_stack[0].weight Parameter containing:
tensor(..., device='meta', size=(3, 3), requires_grad=True)
```

File: functorch/_src/make_functional.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def _get_nested_attr(obj: nn.Module, names: List[str]) -> None:
     if len(names) == 1:
         return getattr(obj, names[0])
     else:
-        _get_nested_attr(getattr(obj, names[0]), names[1:])
+        return _get_nested_attr(getattr(obj, names[0]), names[1:])
```
Comment (zou3519) [inline functorch/_src/make_functional.py:47]: Add a test? (our tests for make_functional are in test/functorch/test_eager_transforms

Comment (srossross) [reply functorch/_src/make_functional.py:47]: Perfect, thanks!

Comment (srossross) [reply functorch/_src/make_functional.py:47]: done in last commit

**Review by zou3519 — APPROVED** [review]
thanks!

**Pull request closed without merging**